### DISCUSSION TITLE
Which XAI methods yield high resolution maps?

### DISCUSSION THREAD

**Discussion by JoanaNRocha** [jacobgil/pytorch-grad-cam#388] [Q&A]
I am looking for XAI methods that output a high resolution map, unlike Grad-CAM, whose heatmap is upsampled to the input image’s size. Which common ones yield high resolution heatmaps? Is that the case of any of the below ones? Thank you in advance.

Integrated Gradients
Saliency
Input X Gradient
Guided Backprop
Deconvolution

**Comment by jacobgil** [comment]
All of them. But I think they are much less informative and prone to errors/misleading than the CAM based methods. 

As a trade off, you can target an earlier layer in the model to get a higher spatial output resolution.
For example, instead of model.layer4, use model.layer3 or 2.

**Reply by JoanaNRocha** [reply]
In what sense do you consider them prone to errors? Thank you for that suggestion!
